ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -511,27 +513,36 @@ prop_only_bootstrap_peers_in_fallback_state ioSimTrace traceNumber =
              . Trace.take traceNumber
              $ ioSimTrace
 
+      consensusModes =
+        map (  snd
+            . head -- ^ the first trace has the consensus mode as it is a static property
+            . Signal.eventsToList
+            . Signal.selectEvents f) events
+      f = (\case
+            DiffusionDebugPeerSelectionTrace (TraceGovernorState _ _ st) -> Just (Governor.consensusMode st)
+            _                                                            -> Nothing)
+
```

Comment:
if you need the static consensus mode just pass it as an argument instead 😃 Also that `head` call is problematic

Reply:
Yes, and no. The consensus mode is static but only for an individual node in the simulation, and we can have a mix. There aren't any tests today that check if Genesis and Praos nodes cooperate, but writing such a test is a possibility with the current setup. So these ConsensusMode values are generated arbitrarily in the diffusion trace for each participant, and it wasn't immediately obvious to me how to extract it out and pass as an argument that's better than this way. But we know that the trace for each node has its initialization parameters as the first trace, and so the call to head here is justified imo.

Reply:
If that's the case I'd prefer you compute a Signal for it instead of a static value. It will be cleaner code and more modular since we can then introduce a script that changes consensus mode and test dynamic changes if needed, e.g.

Reply:
Done